src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -133,11 +137,20 @@ public void initForViewByQuestion(InstructorAttributes instructor,
         Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> questionToResponseMap = bundle.getQuestionResponseMap();
         questionPanels = new ArrayList<InstructorFeedbackResultsQuestionTable>();
         
+        // if there is more than one question, we omit generation of responseRows,
+        // and load them by ajax question by question
+        boolean isLoadingStructureOnly = questionToResponseMap.size() > 1;
+                                        
         for (Map.Entry<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> entry : questionToResponseMap.entrySet()) {
             FeedbackQuestionAttributes question = entry.getKey();
             List<FeedbackResponseAttributes> responses = entry.getValue();
             
-            questionPanels.add(buildQuestionTableAndResponseRows(question, responses, ""));
+            InstructorFeedbackResultsQuestionTable questionPanel = buildQuestionTableAndResponseRows(question, responses, "");
+            if (isLoadingStructureOnly) {
+                questionPanel.setQuestionHasResponses(false);
+            }
```

Comment:
Good catch. Because when loading the structure, the bundle will only have a single response. So for all except one of the question tables, its just the "This question has no responses" message that gets rendered but I realised this behaviour isn't guaranteed.